From 0ba1189cc6c89486e32f996e3cf6f5d941ffbd6c Mon Sep 17 00:00:00 2001
From: Andrei Kholodnyi <andrei.kholodnyi@windriver.com>
Date: Sun, 25 Aug 2024 11:32:01 +0200
Subject: [PATCH 2/4] disable backtrace

---
 .../src/system/backtrace.cpp                        | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/common/autoware_universe_utils/src/system/backtrace.cpp b/common/autoware_universe_utils/src/system/backtrace.cpp
index 7af91bdcc..1b79d6675 100644
--- a/common/autoware_universe_utils/src/system/backtrace.cpp
+++ b/common/autoware_universe_utils/src/system/backtrace.cpp
@@ -16,7 +16,9 @@
 
 #include "rclcpp/rclcpp.hpp"
 
+#if !defined(__VXWORKS__)
 #include <execinfo.h>
+#endif
 
 #include <iostream>
 #include <memory>
@@ -28,16 +30,25 @@ namespace autoware::universe_utils
 
 void print_backtrace()
 {
+#if !defined(__VXWORKS__)
   constexpr size_t max_frames = 100;
   void * addrlist[max_frames + 1];
+#endif
 
+#if !defined(__VXWORKS__)
   int addrlen = backtrace(addrlist, sizeof(addrlist) / sizeof(void *));
-
+#else
+  int addrlen = 0;
+#endif
   if (addrlen == 0) {
     return;
   }
 
+#if !defined(__VXWORKS__)
   char ** symbol_list = backtrace_symbols(addrlist, addrlen);
+#else
+  char ** symbol_list = 0;
+#endif
 
   std::stringstream ss;
   ss << "\n   @   ********** back trace **********" << std::endl;
-- 
2.34.1

